clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -454,7 +454,10 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                   << " type " << memref.instr.type << "\n";
     }
     if (memref.marker.type == TRACE_TYPE_MARKER &&
-        memref.marker.marker_type == TRACE_MARKER_TYPE_TIMESTAMP) {
+        memref.marker.marker_type == TRACE_MARKER_TYPE_TIMESTAMP &&
+        // Timestamps at the start of a chunk should be invisible to analysis tools. These
+        // are duplicate timestamps used to faciliate skipping.
+        shard->prev_entry_.marker.marker_type != TRACE_MARKER_TYPE_CHUNK_FOOTER) {
```

Comment:
This needs to check that the prev_entry_ is actually a marker before looking at a marker type, else you'll get mismatches.